eng/common/scripts/Invoke-GitHubAPI.ps1

```diff
@@ -359,4 +359,52 @@ function Remove-GitHubSourceReferences  {
           -Uri $uri `
           -Headers (Get-GitHubApiHeaders -token $AuthToken) `
           -MaximumRetryCount 3
-}
\ No newline at end of file
+}
+
+function Get-GitHubAuthorizationHeaderFromRepository {
+  param(
+    [Parameter(Mandatory = $true)]
+    $RepoPath
+  )
+
+  try {
+    Push-Location -Path $RepoPath
+
+    $configuration = $(git config --list --local)
```

Comment:
Oh OK, yeah I see what you are getting at now. After doing this investigation into using the API, the limitation is that we need to use:

````yaml
- checkout: self
  persistCredentials: true
```

Given this, using an API call doesn't really provide any value beyond just using the existing Git command approach that Sima came up with (the addition of ```persistCredentials: true``` which is something we would apply to the templates in each language repository.

My thinking is that this can be closed down unless there is some value in retaining the ability to pass out the GitHub token from the configuration file.

Reply:
No. I think we just need to add the presistCredentials option to enable things like the default branch as well as sparse checkout features to work with the private repos.